lib/postgrex/protocol.ex

```diff
@@ -165,12 +156,30 @@ defmodule Postgrex.Protocol do
       {:ok, _} = ret ->
         ret
 
-      {:error, _} = err ->
-        connect_endpoints(remaining_endpoints, sock_opts, timeout, s, status, err)
+      {:error, err} ->
+        connect_endpoints(
+          remaining_endpoints,
+          sock_opts,
+          timeout,
+          s,
+          status,
+          previous_errors ++ [err]
```

Comment:
It is better to prepend and then reverse at the end:

```suggestion
        connect_endpoints(remaining_endpoints, sock_opts, timeout, s, status, [err | previous_errors])
```

It won't matter here, as those lists are likely small, but it is best to adhere to best practices. :)